csrc/kernels.cu

```diff
@@ -1664,6 +1707,9 @@ kOptimizerStatic8bit1StateBlockwise(T* p, T* __restrict__ const g, unsigned char
 										else
 											s1_vals[j] = (s1_vals[j]*beta1) + g_val;
 										break;
+									case LION:
+										s1_vals[j] = s1_vals[j]*beta2 + ((1.0f-beta2)*g_val);
```

Comment:
We can add the following line to have the correct order for Lion:

```c
g_vals[j] = lr*sgn(((float)s1_vals[j])*beta1 + ((1.0f-beta1)*((float)g_vals[j])));
 s1_vals[j] = s1_vals[j]*beta2 + ((1.0f-beta2)*g_val);
```
And then we update line 1752 instead with:
```c
p_vals[j] = (float)p_vals[j] - (float)g_vals[j];
```